src/core/meshcop/dataset_manager.cpp

```diff
@@ -601,11 +601,8 @@ otError DatasetManager::SendSetRequest(const otOperationalDataset &aDataset, con
     VerifyOrExit((message = NewMeshCoPMessage(mNetif.GetCoap(), header)) != NULL, error = OT_ERROR_NO_BUFS);
 
 #if OPENTHREAD_ENABLE_COMMISSIONER && OPENTHREAD_FTD
-    bool isCommissioner;
 
-    isCommissioner = mNetif.GetCommissioner().GetState() != kCommissionerStateDisabled ? true : false;
-
-    if (isCommissioner)
+    if (mNetif.GetCommissioner().IsActive())
```

Comment:
@jwhui, the earlier code was checking if we state is anything other than ` Disabled` to set `isCommisioner`...New code is checking if state is exactly `ACTIVE` (so it blocks if we get here with state being `PETITION`)...

I assume the new code is what we want and the intended behavior.... Correct?